sdk/core/System.ClientModel/src/ModelReaderWriter/SequenceBufferReader.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.ClientModel.Internal;
+
+internal class SequenceBufferReader : IDisposable
+{
+    private readonly SequenceBuffer[] _buffers;
+    private readonly int _count;
+    private volatile int _isDisposed;
+    private volatile int _copyCount;
+
+    public SequenceBufferReader(SequenceBuffer[] buffers, int count)
+    {
+        _buffers = buffers;
+        _count = count;
+    }
+
+    public bool TryComputeLength(out long length)
+    {
+        if (_isDisposed == 1)
```

Comment:
Why is it an int? Also, I think comparing to zero is a bit faster.

Reply:
It is an int so I can Interlock.Exchange when we enter the dispose method.  https://github.com/Azure/azure-sdk-for-net/pull/41560/files#diff-db6a220b45cbe3e2b65a7fb858a7a752f22ce8e989616489c90fae6ab18a52e1R120